File: week26/script_test_lamw.py
from cmdstanpy import CmdStanModel

N = 1000
simu_model = CmdStanModel(stan_file='/Users/nshah/work/gsoc/week26/lamw_simu.stan')
simu_fit = simu_model.sample(data={'mu': 0.5, 'sigma': 2, 'delta': 1}, chains=1, iter_sampling=N, iter_warmup=0, fixed_param=True);
y = simu_fit.draws_pd()['y'].values

infer_model = CmdStanModel(stan_file='/Users/nshah/work/gsoc/week26/lamw_infer.stan')
infer_fit = infer_model.sample(data={'y': y, 'N': N}, chains=4);
infer_fit.diagnose()
infer_fit.summary()

File: week27/script_test_lamw.R
library(cmdstanr)
set_cmdstan_path('/Users/nshah/cmdstan')
fp <- file.path("./lambertw.stan")

# ------------------

N <- 1000
mu <- 0
sigma <- 1
delta <- 1

x <- rnorm(N, mean = mu, sd = sigma)
x_hat <- (x - mu)/sigma
y <- x_hat * exp(delta/2 * x_hat^2) * sigma + mu;

mod <- cmdstan_model(fp, force_recompile = T)
mod_out <- mod$sample(
    data = list(N = N, y = y),
    chains = 2,
    init = 5,
    adapt_delta = 0.8,
    parallel_chains = 2,
    iter_warmup = 500,
    iter_sampling = 200
)

mod_out$summary()

# ------------------

library(LambertW)
yy = LambertW::rLambertW(N, beta = c(mu, sigma), delta = delta, distname = "normal")
